lib/ace/mode/graphqlscheme_highlight_rules.js

```diff
@@ -0,0 +1,93 @@
+/* ***** BEGIN LICENSE BLOCK *****
+ * Distributed under the BSD license:
+ *
+ * Copyright (c) 2010, Ajax.org B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Ajax.org B.V. nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL AJAX.ORG B.V. BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * ***** END LICENSE BLOCK ***** */
+
+define(function(require, exports, module) {
+"use strict";
+
+var oop = require("../lib/oop");
+var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
+
+var GraphQLSchemeHighlightRules = function() {
+
+    var keywords = [
+      "type",
+      "interface",
+      "union",
+      "enum",
+      "schema",
+      "input",
+      "implements",
+      "extends",
+      "scalar"
+    ].join("|");
+
+//     var builtinConstants = (
+//         "true|false"
+//     );
+//
+//     var builtinFunctions = (
+//         "avg|count|first|last|max|min|sum|ucase|lcase|mid|len|round|rank|now|format|" +
+//         "coalesce|ifnull|isnull|nvl"
+//     );
+//
+    var dataTypes = [
+        "Int","Float", "String", "ID", "Boolean"
+    ].join("|");
+
+    var keywordMapper = this.createKeywordMapper({
+        // "support.function": builtinFunctions,
+        "keyword": keywords,
+        // "constant.language": builtinConstants,
+        "storage.type": dataTypes
+    }, "identifier");
+
+    this.$rules = {
+      "start" : [ {
+        token : "comment",
+        regex : "#.*$"
+      }, {
+        token : "paren.lparen",
+        regex : /[\[({]/,
+        next  : "start"
+      }, {
+        token : "paren.rparen",
+        regex : /[\])}]/
```

Comment:
how is graphql normally used? I could only find examples for client operation.
I suggest naming the mode graphql based on what other editors do, but i may be missing something since i am not familiar with the language.